maps_dashboards/public/components/app.tsx

```diff
@@ -17,20 +17,44 @@ interface MapsDashboardsAppDeps {
   notifications: CoreStart['notifications'];
   http: CoreStart['http'];
   navigation: NavigationPublicPluginStart;
+  savedObjectsClient: CoreStart['savedObjects']['client'];
+  setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'];
 }
 
-export const MapsDashboardsApp = ({ basename, notifications, http }: MapsDashboardsAppDeps) => {
+export const MapsDashboardsApp = ({
+  basename,
+  notifications,
+  http,
+  savedObjectsClient,
+  navigation,
+  setHeaderActionMenu,
```

Comment:
The common pattern followed in plugins to pass down these services to any component that needs them is using `OpenSearchDashboardsContextProvider` e.g. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/vis_builder/public/application/index.tsx#L24. You can then use the services anywhere in the app using the `useOpenSearchDashboards` hook e.g. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/vis_builder/public/application/components/workspace.tsx#L23-L29

Reply:
Thanks! The updated code used the `OpenSearchDashboardsContextProvider ` and `useOpenSearchDashboards ` for the whole app.